docs/fundamentals/networking/http/httpclient-migrate-from-httpwebrequest.md

```diff
@@ -0,0 +1,468 @@
+        if (cacheItem.Addresses.Length == 1)
+        {
+            connectAddress = cacheItem.Addresses[0];
+        }
+        else
+        {
+            int index = cacheItem.Index;
+            connectAddress = cacheItem.Addresses[index];
+            cacheItem.IncreaseIndex();
+        }
+
+        var socket = new Socket(SocketType.Stream, ProtocolType.Tcp)
+        {
+            NoDelay = true
+        };
+        try
+        {
+            await socket.ConnectAsync(connectAddress, context.DnsEndPoint.Port, cancellationToken);
+            return new NetworkStream(socket, ownsSocket: true);
+        }
+        catch
+        {
+            socket.Dispose();
+            throw;
+        }
+    }
+};
+var client = new HttpClient(handler);
+var response = await client.GetAsync(uri);
+
+class DnsCache(IPAddress[] addresses, DateTime cacheExpireTime, int index)
+{
+    public IPAddress[] Addresses { get; } = addresses;
+    public DateTime CacheExpireTime { get; } = cacheExpireTime;
+    public int Index { get; private set; } = index;
+
+    public void IncreaseIndex()
+    {
+        Index = (Index + 1) % Addresses.Length;
+    }
+}
+```
+
+### Example: Setting SocketsHttpHandler Properties
+
+TODO:
+
+### Example: Change ImpersonationLevel
+
+TODO:
+
+## Usage of Certificate and TLS Related Properties in HttpClient
+
+When working with `HttpClient`, you may need to handle client certificates for various purposes, such as custom validation of server certificates or fetching the server certificate. `HttpClient` provides several properties and options to manage certificates effectively.
+
+### Example: Enabling CRL Check with SocketsHttpHandler
+
+The `CheckCertificateRevocationList` property in `SocketsHttpHandler` allows developers to enable or disable the check for certificate revocation lists (CRL) during SSL/TLS handshake. Enabling this property ensures that the client verifies whether the server's certificate has been revoked, enhancing the security of the connection.
+
+**Old Code Using `HttpWebRequest`**:
+
+```csharp
+ServicePointManager.CheckCertificateRevocationList = true;
+HttpWebRequest request = (HttpWebRequest)WebRequest.Create(uri);
+HttpWebResponse response = (HttpWebResponse)request.GetResponse();
+```
+
+**New Code Using `HttpClient`**:
+
+```csharp
+bool checkCertificateRevocationList = true;
+var handler = new SocketsHttpHandler
+{
+    SslOptions =
+    {
+        CertificateRevocationCheckMode = checkCertificateRevocationList ? X509RevocationMode.Online : X509RevocationMode.NoCheck,
+    }
+};
+var client = new HttpClient(handler);
+var response = await client.GetAsync(uri);
+```
+
+### Example: Fetch Certificate
+
+To fetch the certificate from the RemoteCertificateValidationCallback in HttpClient, you can use the ServerCertificateCustomValidationCallback property of HttpClientHandler or SocketsHttpHandler. This callback allows you to inspect the server's certificate during the SSL/TLS handshake.
```

Comment:
It's actually different callback with different shape, so either I'd only mention the `SocketsHttpHandler.SslOptions` one, or split the new code into 2 examples.

NITs:
```suggestion
To fetch the certificate from the `RemoteCertificateValidationCallback` in `HttpClient`, you can use the `ServerCertificateCustomValidationCallback` property of `HttpClientHandler` or `SocketsHttpHandler.SslOptions`. This callback allows you to inspect the server's certificate during the SSL/TLS handshake.
```